### PR TITLE
feat/PN-12582: openAPI definition for PG public and virtual keys

### PR DESCRIPTION
**Pull request by mflauti**
## Short description
This PR includes the definition of the openAPI needed to allow PG recipients to create and manage public and virtual keys which can be by their own system as auth mechanism

## List of changes proposed in this pull request
- api-internal-pn-bff-pg-apikeys.yaml file which contains the openAPI definition
- virtual and public keys schemas

### CODE REVIEW

File: docs/openapi/api-internal-pn-bff-pg-apikeys.yaml [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+              schema:
+                $ref: './common-refs.yaml#/components/schemas/Problem'
+        '500':
+          description: Internal error
+          content:
+            application/problem+json:
+              schema:
+                $ref: './common-refs.yaml#/components/schemas/Problem'
+
+  "/virtual-keys/{id}":
```
Comment (AndreaCimini90) [inline docs/openapi/api-internal-pn-bff-pg-apikeys.yaml:372]: add `/bff/v1`

File: docs/openapi/api-internal-pn-bff-pg-apikeys.yaml [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+          description: Internal error
+          content:
+            application/problem+json:
+              schema:
+                $ref: './common-refs.yaml#/components/schemas/Problem'
+
+components:
+  parameters:
+    ### PARAMETRI RICERCA PUBLIC KEY
+    pathKid:
```
Comment (AndreaCimini90) [inline docs/openapi/api-internal-pn-bff-pg-apikeys.yaml:467]: pathKid and pathVirtualKeyId are the same. merge them into pathKid

File: docs/openapi/api-internal-pn-bff-pg-apikeys.yaml [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+      description: >-
+        Identificativo univoco della public key
```
Comment (AndreaCimini90) [inline docs/openapi/api-internal-pn-bff-pg-apikeys.yaml:469]: After merging chose a generic description

File: docs/openapi/publicKeys-schemas/public-keys.yaml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+          description: Data censimento public key
+          format: date-time
+        status:
+          $ref: '#/components/schemas/BffPublicKeyStatus'
+        statusHistory:
+          type: array
+          description: Storico degli stati delle public key
+          items:
+            $ref: '#/components/schemas/BffPublicKeyStatusHistory'
+    BffPublicKeyStatus:
```
Comment (AndreaCimini90) [inline docs/openapi/publicKeys-schemas/public-keys.yaml:50]: add empty lines between model definitions

File: docs/openapi/publicKeys-schemas/public-keys.yaml [hunk 5]
```diff
@@ -0,0 +1,95 @@
+info:
```
Comment (AndreaCimini90) [inline docs/openapi/publicKeys-schemas/public-keys.yaml:1]: are we planning to change the response that comes to us from the microservice? if no, we don't need to redefine all the models.
Check which models will change and redefine only them

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: docs/openapi/api-internal-pn-bff-pg-apikeys.yaml [hunk 6]
```diff
@@ -0,0 +1,495 @@
...
+                $ref: './common-refs.yaml#/components/schemas/Problem'
+        '500':
+          description: Internal error
+          content:
+            application/problem+json:
+              schema:
+                $ref: './common-refs.yaml#/components/schemas/Problem'
+
+  "/bff/v1/pg/public-key":
+    post:
```
[the remark this answers is not in the document]
Comment (mflauti) [reply docs/openapi/api-internal-pn-bff-pg-apikeys.yaml:90]: This definition was made mimicking the api that the bff is designed to consume. In that api these 2 paths are kept separated, but I can change it if you think it's preferable.

File: docs/openapi/api-internal-pn-bff-pg-apikeys.yaml (hunk 2 above)
Comment (mflauti) [reply docs/openapi/api-internal-pn-bff-pg-apikeys.yaml:467]: Even though their type is the same (at least right now) they conceptually refer respectively to the public and virtual key id, that's why I preferred to keep them separated. Anyway, if you think that's a useless redundancy I can merge them in a single definition.

File: docs/openapi/api-internal-pn-bff-pg-apikeys.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (AndreaCimini90) [reply docs/openapi/api-internal-pn-bff-pg-apikeys.yaml:90]: yes we don't have to copy the structure of the micro-service api. here we merge the two methods under the same path to make it similar to the others api

File: docs/openapi/api-internal-pn-bff-pg-apikeys.yaml [hunk 7]
```diff
@@ -0,0 +1,495 @@
...
+                $ref: './common-refs.yaml#/components/schemas/Problem'
+        '500':
+          description: Internal error
+          content:
+            application/problem+json:
+              schema:
+                $ref: './common-refs.yaml#/components/schemas/Problem'
+
+  "/bff/v1/pg/public-key/{kid}/delete":
+    put:
```
[the remark this answers is not in the document]
Comment (AndreaCimini90) [reply docs/openapi/api-internal-pn-bff-pg-apikeys.yaml:132]: so it is a status change?

[the remark this answers is not in the document]
Comment (mflauti) [reply docs/openapi/api-internal-pn-bff-pg-apikeys.yaml:132]: it's essentially a change to one of key's properties

File: docs/openapi/publicKeys-schemas/public-keys.yaml [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  version: v1.0
+components:
+  schemas:
+
+    BffPublicKeysResponse:
```
Comment (AndreaCimini90) [inline docs/openapi/publicKeys-schemas/public-keys.yaml:6]: Check if there are unused models and remove them

File: docs/openapi/virtualKeys-schemas/virtual-keys.yaml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+info:
+  version: v1.0
+components:
+  schemas:
+    BffVirtualKeysResponse:
```
Comment (AndreaCimini90) [inline docs/openapi/virtualKeys-schemas/virtual-keys.yaml:5]: Check if there are unused models and remove them

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**